frontend/src/App.tsx

```diff
@@ -215,15 +229,20 @@ function appStateReducer(state: CoveyAppState, update: CoveyAppUpdate): CoveyApp
       nextState.gameStarted = true;
       break;
     case 'disconnect':
+      state.socket?.emit('finishGame', state.myPlayerID, -1, true);
       state.socket?.disconnect();
+      closedInstructions = false;
       return defaultAppState();
     case 'toggleQuit':
       nextState.toggleQuit = !state.toggleQuit;
```

Comment:
can you add a check here in `toggleQuit` to only actually toggle the state if `gameStatus` == `inGame` and/or `gameStarted`? right now we can call it whenever since we implemented it before `gameStatus` ^^;;